pkg/util/xorm/session_insert.go

```diff
@@ -780,3 +797,34 @@ func (session *Session) insertMap(columns []string, args []any) (int64, error) {
 	}
 	return affected, nil
 }
+
+type snowflake struct {
+	mu       sync.Mutex
+	nodeID   int64
+	sequence int64
+	lastTime int64
+}
+
+// NewSnowflake creates a new instance with a random node ID (0-1023)
+const snowflakeEpoch = 1288834974657       // 2010-11-04 01:42:54.657 UTC
+func newSnowflake(nodeID int64) *snowflake { return &snowflake{nodeID: nodeID & 0x3ff} }
+
+func (s *snowflake) Generate() int64 {
```

Comment:
This code seems to be copied from somewhere. Shall we copy its test too, and mention the original source (for possible updates in the future)?